src/main/java/io/split/android/client/service/sseclient/PushNotificationManager.java

```diff
@@ -67,15 +73,22 @@ private void resetSseKeepAliveTimer() {
         mResetSseKeepAliveTimerTaskId = mTaskExecutor.schedule(
                 new SseKeepAliveTimer(),
                 SSE_RECONNECT_TIME_IN_SECONDS,
-                this);
+                null);
+    }
+
+    private void resetSseTokenExpiredTimer(long expirationTime) {
+        mSseTokenExpiredTimerTaskId = mTaskExecutor.schedule(
+                new SseTokenExpiredTimer(),
+                expirationTime - System.currentTimeMillis() / 1000,
```

Comment:
recheck. I think `expirationTime` (`exp` in token) is in seconds, not millis. Also we should reschedule the task to run before the expiration, something like:`expirationTime - System.currentTimeMillis() - TIME_BEFORE_EXPIRATION`

Reply:
Doc, You're right about TIME_BEFORE_EXPIRATION. Gonna change that.
Thanks!